pkg/project/resource_project.go

```diff
@@ -217,9 +218,20 @@ func projectResource() *schema.Resource {
 		},
 
 		"repos": {
-			Type:        schema.TypeSet,
-			Optional:    true,
-			Elem:        &schema.Schema{Type: schema.TypeString},
+			Type:     schema.TypeSet,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			MinItems: 0,
+			MaxItems: func() int {
+				rlv := os.Getenv("REPO_LIMIT_OVERRIDE")
+				iRlv, err := strconv.Atoi(rlv)
```

Comment:
I am not sure what `iRlv` means. Just wrap the `os.Getenv` in the conversion and call it `override` - By wrapping it directly you can skip the `""` check, I think. What will the parse do on `""` ? Give 0 or error?

So you can just have:
```go
override, err := strconv.Atoi(os....)
if err != nil {
  override = 0 // an error means bad parse 
}
return math.Max(override,100)
```

Also, look at this, which is even simpler!

https://stackoverflow.com/questions/40326540/how-to-assign-default-value-if-env-var-is-empty

```go
func getEnv(key, fallback string) string {
    value, exists := os.LookupEnv(key)
    if !exists {
        value = fallback
    }
    return value
}
```

Reply:
improved.